backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/testrepository/ShopRepositoryTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.google.gpay.virtualqueue.backendservice.testrepository;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.gpay.virtualqueue.backendservice.entities.Shop;
+import com.google.gpay.virtualqueue.backendservice.respository.ShopRepository;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@SpringBootTest
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = ShopRepository.class)
+public class ShopRepositoryTest {
+    ShopRepository shopRep = new ShopRepository();
+
+    @BeforeEach
+    public void setUp() {
+        shopRep.token_mappings.clear();
+        shopRep.shop_mappings.clear();
+        shopRep.last_id = 1;
+    }
+
+    @Test
+    public void testAddShop() throws Exception {
+        Shop shop = new Shop("uid", "name", "shop1", "address", "1111111111", "type");
+        String result = shopRep.CreateShop(shop);
+        int size = shopRep.shop_mappings.size();
+        assertEquals("size ", 1, size);
```

Comment:
Try to have your tests in the three blocks separate by a newline each: go/unit-test-practices#structure.

Reply:
The link doesn't work but I am still looking in it.

Reply:
go/interns2020-testing
See if this helps Ishita. (for Unit testing)

Reply:
Yes I have seen that but it doesn't talk about arrange act assert blocks that are needed here. Still thanks a lot for sharing!